api/publish.go

```diff
@@ -347,12 +431,28 @@ func apiPublishUpdateSwitch(c *gin.Context) {
 		published.MultiDist = *b.MultiDist
 	}
 
-	resources = append(resources, string(published.Key()))
-	taskName := fmt.Sprintf("Update published %s (%s): %s", published.SourceKind, strings.Join(updatedComponents, " "), strings.Join(updatedSnapshots, ", "))
+	revision := published.ObtainRevision()
+	sources := revision.Sources
+
+	if published.SourceKind == deb.SourceSnapshot {
+		for _, snapshotInfo := range b.Snapshots {
+			component := snapshotInfo.Component
+			name := snapshotInfo.Name
+			sources[component] = name
```

Comment:
is `sources` used somewhere ?

Reply:
Yes in line 441 :-).

Reply:
sorry, I don't get it :-)

the sources variable (in 435)  is only assigned (in 441), and not used below, or do I miss something ?

Reply:
`sources` gets defined and set in line 435. The variable is within the loop in line 441.

Reply:
ah, it is setting the revision.Sources, adding the components from the Snapshots argument. maybe this deserves a comment ;-)